lib/elixir/lib/calendar/iso.ex

```diff
@@ -663,6 +664,147 @@ defmodule Calendar.ISO do
     end
   end
 
+  @doc """
+  Parses an ISO 8601 formatted duration string to a list of `Duration` compabitble unit pairs.
+
+  """
+  @doc since: "1.17.0"
+  @spec parse_duration(String.t()) :: {:ok, [Duration.unit_pair()]} | {:error, atom()}
+  def parse_duration("P" <> string) do
+    parse_date_duration(string, %{}, "")
```

Comment:
I think the implementation is wrong because it allows YDMW in any other, and then HMS in any order? You'd need several small functions, parse_year_duration, parse_month_duration, etc....

Reply:
That's correct, i found other implementations are also not strict about the order, but i agree we want to be strict about it here. I should find some time in the coming days to rewrite this